plonky2/src/gadgets/curve_msm.rs

```diff
@@ -0,0 +1,133 @@
+use num::BigUint;
+use plonky2_field::extension_field::Extendable;
+
+use crate::curve::curve_types::{Curve, CurveScalar};
+use crate::field::field_types::Field;
+use crate::gadgets::biguint::BigUintTarget;
+use crate::gadgets::curve::AffinePointTarget;
+use crate::hash::hash_types::RichField;
+use crate::hash::keccak::KeccakHash;
+use crate::plonk::circuit_builder::CircuitBuilder;
+use crate::plonk::config::{GenericHashOut, Hasher};
+
+impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
+    /// Computes `n*p + m*q` using windowed MSM, with a 2-bit window.
+    /// See Algorithm 9.23 in Handbook of Elliptic and Hyperelliptic Curve Cryptography for a
+    /// description.
+    /// Note: Doesn't work if `p == q`.
+    pub fn curve_msm<C: Curve>(
+        &mut self,
+        p: &AffinePointTarget<C>,
+        q: &AffinePointTarget<C>,
+        n: &BigUintTarget,
```

Comment:
Should we make it consistent with `curve_scalar_mul`? I feel like both should take a biguint target, or both should take a scalar field nonnative target?

Reply:
Good idea, done.